localstack-core/localstack/services/stepfunctions/asl/component/intrinsic/argument/argument.py

```diff
@@ -0,0 +1,92 @@
+import abc
+from typing import Any, Final, Optional
+
+from localstack.services.stepfunctions.asl.component.common.string.string_expression import (
+    StringVariableSample,
+)
+from localstack.services.stepfunctions.asl.component.eval_component import EvalComponent
+from localstack.services.stepfunctions.asl.eval.environment import Environment
+from localstack.services.stepfunctions.asl.utils.json_path import extract_json
+
+
+class Argument(EvalComponent, abc.ABC):
+    @abc.abstractmethod
+    def _eval_argument(self, env: Environment) -> Any: ...
+
+    def _eval_body(self, env: Environment) -> None:
+        argument = self._eval_argument(env=env)
+        env.stack.append(argument)
+
+
+class ArgumentLiteral(Argument):
```

Comment:
These Argument* implementations seem identical apart from inheriting the new Argument abstraction.
I think they are easier to understand them grouped here rather than spread across individual files 👍